Remove hardcoded constants in SD sample

yatarkan opened the pull request.

yatarkan committed:
Add latest optimum-intel to requirements

yatarkan committed:
Remove deprecated cli argument

yatarkan committed:
Remove hardcoded constants from SD sample

yatarkan committed:
Add static cast

yatarkan committed:
Move setting conda env after dependencies installation

yatarkan committed:
Remove redundantcli  argument

yatarkan committed:
Revert "Move setting conda env after dependencies installation"

This reverts commit d4142bcb42d3a41f7d514fbf9e86db35b45b2fab.

yatarkan committed:
Specify optimum intel branch in requirements

yatarkan committed:
Merge branch 'master' into yt/remove-constants-sd-sample

yatarkan committed:
Instlall git with conda in GH workflow

nikita-malininn committed:
Update NNCF default configs (#419)

**Updated NNCF defaults for:**

- dolly-v2-3b
- opt-2.7b
- red-pajama-incite-chat-3b-v1
- vicuna-7b-v1.5
- stablelm-tuned-alpha-3b

Based on the experiments from 135227

ilya-lavrenov commented on image_generation/stable_diffusion_1_5/cpp/requirements.txt:
```diff
@@ -2,5 +2,5 @@
 diffusers==0.27.2
 transformers==4.39.3
-optimum-intel[nncf,openvino]==1.16.0
+optimum-intel[nncf,openvino] @ git+https://github.com/huggingface/optimum-intel.git@main
```
why do we need latest optimum?

ilya-lavrenov commented on image_generation/stable_diffusion_1_5/cpp/src/main.cpp:
```diff
@@ -397,10 +396,11 @@ int32_t main(int32_t argc, char* argv[]) try {
 
         // latents are multiplied by 'init_noise_sigma'
-        ov::Shape latent_shape = noise.get_shape(), latent_model_input_shape = latent_shape;
+        ov::Shape latent_shape = ov::Shape({batch_size, static_cast<size_t>(sample_shape[1].get_length()), height / VAE_SCALE_FACTOR, width / VAE_SCALE_FACTOR});
```
nitpick : should we extract `sample_shape[1].get_length()` as variable `unet_in_channels` to denote relationship between dimensions? 
Otherwise, people don't know why this dimension is taken..

Wovchena committed:
Split convert_tokenizer and greedy_causal_lm, ubuntu-20.04-4-cores (#420)

cpp-greedy_causal_lm-redpajama-3b-chat randomply fails. Split the
commands to find whether it's convert_tokenizer or runtime

ilya-lavrenov commented on image_generation/stable_diffusion_1_5/cpp/requirements.txt:
```diff
@@ -2,5 +2,5 @@
 diffusers==0.27.2
 transformers==4.39.3
-optimum-intel[nncf,openvino]==1.16.0
+optimum-intel[nncf,openvino] @ git+https://github.com/huggingface/optimum-intel.git@main
```
I think we can try https://github.com/huggingface/optimum-intel/pull/697 and merge if it helps

ilya-lavrenov replied:
you can proceed with current PR, because optimum is updated here https://github.com/openvinotoolkit/openvino.genai/pull/427

Now, `convert_tokenizers` is not required

as-suvorov committed:
Consume all draft tokens for speculative decoding (#424)

Ticket: 140110

Wovchena committed:
Merge releases/2024/1 into master (#427)

Co-authored-by: guozhong wang <guozhong.wang@intel.com>
Co-authored-by: Chen Peter <peter.chen@intel.com>
Co-authored-by: yatarkan <yaroslav.tarkan@intel.com>

Wovchena committed:
Add hardware suggestion to samples (#394)

Ticket 139548

dependabot[bot] committed:
Bump einops from 0.7.0 to 0.8.0 in /text_generation/causal_lm/cpp (#430)

Bumps [einops](https://github.com/arogozhnikov/einops) from 0.7.0 to
0.8.0.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/arogozhnikov/einops/releases">einops's
releases</a>.</em></p>
<blockquote>
<h2>v0.8.0: tinygrad, small fixes and updates</h2>
<h2>TLDR</h2>
<ul>
<li>tinygrad backend added</li>
<li>resolve warning in py3.11 related to docstring</li>
<li>remove graph break for unpack</li>
<li><strong>breaking</strong> TF layers were updated to follow new
instructions, new layers compatible with TF 2.16, and not compatible
with old TF (certainly does not work with TF2.13)</li>
</ul>
<h2>What's Changed</h2>
<ul>
<li>Fix invalid escape sequence in einsum docstring by <a
href="https://github.com/atwam"><code>@​atwam</code></a> in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/298">arogozhnikov/einops#298</a></li>
<li>Tinygrad support by <a
href="https://github.com/blueridanus"><code>@​blueridanus</code></a> in
<a
href="https://redirect.github.com/arogozhnikov/einops/pull/297">arogozhnikov/einops#297</a></li>
<li>Coerce bool to int in unpack by <a
href="https://github.com/drubinstein"><code>@​drubinstein</code></a> in
<a
href="https://redirect.github.com/arogozhnikov/einops/pull/287">arogozhnikov/einops#287</a></li>
<li>Remove oneflow from testing by <a
href="https://github.com/arogozhnikov"><code>@​arogozhnikov</code></a>
in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/289">arogozhnikov/einops#289</a></li>
<li>tests: fix torch installation to force CPU by <a
href="https://github.com/arogozhnikov"><code>@​arogozhnikov</code></a>
in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/288">arogozhnikov/einops#288</a></li>
<li>Allow anonymous axes in parse_shape, fix <a
href="https://redirect.github.com/arogozhnikov/einops/issues/302">#302</a>
by <a
href="https://github.com/arogozhnikov"><code>@​arogozhnikov</code></a>
in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/303">arogozhnikov/einops#303</a></li>
<li>Codebase standards + update TF layers by <a
href="https://github.com/arogozhnikov"><code>@​arogozhnikov</code></a>
in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/318">arogozhnikov/einops#318</a></li>
<li>update github actions by <a
href="https://github.com/arogozhnikov"><code>@​arogozhnikov</code></a>
in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/319">arogozhnikov/einops#319</a></li>
</ul>
<h2>New Contributors</h2>
<ul>
<li><a
href="https://github.com/drubinstein"><code>@​drubinstein</code></a>
made their first contribution in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/287">arogozhnikov/einops#287</a></li>
<li><a href="https://github.com/atwam"><code>@​atwam</code></a> made
their first contribution in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/298">arogozhnikov/einops#298</a></li>
<li><a
href="https://github.com/blueridanus"><code>@​blueridanus</code></a>
made their first contribution in <a
href="https://redirect.github.com/arogozhnikov/einops/pull/297">arogozhnikov/einops#297</a></li>
</ul>
<p><strong>Full Changelog</strong>: <a
href="https://github.com/arogozhnikov/einops/compare/v0.7.0...v0.8.0">https://github.com/arogozhnikov/einops/compare/v0.7.0...v0.8.0</a></p>
</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/arogozhnikov/einops/commit/fe9d81d577dec5a224d572a5bc7d44ab8bd914eb"><code>fe9d81d</code></a>
bump version to 0.8.0</li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/10e19b4bd633a3e4de9bc4cafcad57f4ed66d0ef"><code>10e19b4</code></a>
prepare readme for a new release</li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/b67cfea95639652265030df1bcdb401feacaf242"><code>b67cfea</code></a>
Rename KerasBackend -&gt; TFKerasBackend to avoid confusion with keras
3.</li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/f33113e46995aabbbd2c59683ce1a6c77174c62e"><code>f33113e</code></a>
update github actions</li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/cff63f2dc859fd76309dbafff77b465be90ee71e"><code>cff63f2</code></a>
Codebase standards + update TF layers (<a
href="https://redirect.github.com/arogozhnikov/einops/issues/318">#318</a>)</li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/2ab5c2a5f693648a16eb3ceb8d2e1ed857101bde"><code>2ab5c2a</code></a>
switch to stable paddlepaddle in testing, because of <a
href="https://github.com/Paddl">https://github.com/Paddl</a>...</li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/60b87dee0464ebb6def063a7e4cd44cfcbd18d3c"><code>60b87de</code></a>
remove announcement</li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/5655ce12ce49dac3fe6402b510d09f6bb0715b5b"><code>5655ce1</code></a>
Merge pull request <a
href="https://redirect.github.com/arogozhnikov/einops/issues/303">#303</a>
from arogozhnikov/dev</li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/5b41d90142917744172e308d5bf25a606413cb43"><code>5b41d90</code></a>
allow anonymous axes in parse_shape, fix <a
href="https://redirect.github.com/arogozhnikov/einops/issues/302">#302</a></li>
<li><a
href="https://github.com/arogozhnikov/einops/commit/9a9b30432aaceac7b35077a5e19bfa26c8c12ef4"><code>9a9b304</code></a>
delete experimental data-api_packing as production version is available
now</li>
<li>Additional commits viewable in <a
href="https://github.com/arogozhnikov/einops/compare/v0.7.0...v0.8.0">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=einops&package-manager=pip&previous-version=0.7.0&new-version=0.8.0)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

yatarkan committed:
Move unet in channels to separate var

yatarkan committed:
Merge branch 'master' into yt/remove-constants-sd-sample

ilya-lavrenov commented:
Please, resolve conflicts
